ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -219,6 +221,16 @@ private void executeRounds(
           Preconditions.checkState(fetchResult.getResult() == RoutingResult.SUCCESS);
           processedPartitions++;
         }
+      } catch (TimeoutException timeoutException) {
+        final MaterializationException exception =
+            new MaterializationException(
+            "Query timeout exception");
+        for (Entry<Integer, List<Exception>> entry: exceptionsPerPartition.entrySet()) {
+          for (Exception excp: entry.getValue()) {
+            exception.addSuppressed(excp);
+          }
+        }
+        throw exception;
```

Comment:
is this something we could easily build a test for? (I guess re my first comment about configuration would make it easier to test)